Import demo components from @storybook/react

shilman opened the pull request:
Issue: #1266

## What I did

Import demo components from `@storybook/react` instead of `@storybook/components`

We do this so we can have a uniform way to do this across frameworks
(e.g. RN, vue)

## How to test

To make sure the deps work:
```
yarn bootstrap
cd examples/cra-kitchen-sink
yarn storybook
```

Does not test `getstorybook`, but I think we can assume that will work (but please review)

shilman committed:
Export demo components from @storybook/react

We do this so we can have a uniform way to do this across frameworks
(e.g. RN, vue)

shilman committed:
Update cra-kitchen-sink to use demo imports from `@storybook/react`

usulpro reviewed (dismissed):
works fine for me!

alexandrebodin reviewed (approved):
it works

shilman committed:
Move demo components from `@storybook/components` to `@storybook/react`

shilman committed:
Minor cleanup

UsulPro commented:
still WIP? need to remove `\lib\components\src\demo`

shilman committed:
Remove demo components from `@storybook/components`

shilman commented:
Thanks @usulpro . I forgot to remove them earlier

shilman committed:
Add a file to `@storybook/components` so that the build doesn't break

shilman commented:
@ndelangen @UsulPro @alexandrebodin a test is failing but I don't have time to figure it out 😭 

```
 FAIL  examples/test-cra/src/storyshots.test.js
 Test suite failed to run

Your test suite must contain at least one test.
```

alexandrebodin reviewed (commented):
We could create a demo.js in the root folder to reexport the components from dist ? 

I don't like that user can access filles like this :( 

Juste a preference no big deal for me ;)

shilman commented:
Need to be offline for next four days. Would ❤️ if somebody could get this merged and shipped. @Norbert?

UsulPro commented on addons/storyshots/stories/directly_required/index.js:
```diff
@@ -3,7 +3,7 @@ import React from 'react';
 import { action } from '@storybook/addon-actions'; // eslint-disable-line
 
-import Button from '@storybook/components/dist/demo/Button';
+import Button from '@storybook/react/dist/demo/Button';
```
investigated that test fails after this change
not sure what the reason, possible someone who more familiar with storyshots could help?

tmeasday replied:
FYI these particular files are not tested by CI

UsulPro replied:
it fails at [examples/test-cra/src/storyshots.test.js](https://travis-ci.org/storybooks/storybook/builds/245331027#L696) but after this change. 
[#2833 passed](https://travis-ci.org/storybooks/storybook/builds/244077064)
[#2836 failed](https://travis-ci.org/storybooks/storybook/builds/244080125)

but possible it's more CI issue then this PR?

tmeasday replied:
I'm not sure why it sometimes passes to be honest, I would have thought it would consistently fail due to: https://github.com/storybooks/storybook/pull/1303#issuecomment-310052041

ndelangen committed:
Merge branch 'master' into 1266-fix-components-deps

ndelangen commented:
Looks good to me, what's failing the build?

tmeasday commented on app/react-native/package.json:
```diff
@@ -28,7 +28,7 @@
     "@storybook/addons": "^3.1.2",
     "@storybook/channel-websocket": "^3.1.2",
-    "@storybook/components": "^3.0.0",
+    "@storybook/react": "^3.0.0",
```
Not sure we should be doing this

tmeasday replied:
Does RN use these components? AFAIK not (if so it should be changed in this PR).

tmeasday commented:
Cf. https://github.com/storybooks/storybook/issues/1286 -- not sure this test passes on any branch with npm5

shilman committed:
Remove unnecessary `@storybook/react` dep in `@storybook/react-native`

shilman commented:
Currently the build is broken because `test-cra` requires `@storybook/addon-storyshots` AND `@storybook/react`. At the same time `@storybook/addon-storyshots` requires `@storybook/react`.

When we run the `test-cra` tests, two versions of `@storybook/react` are getting loaded, and this leads to a test error “Your test suite must contain at least one test.”

If we remove `@storybook/react` from `@storybook/addon-storyshots`'s `package.json` then the tests pass again.

However, we are using demo components from `@storybook/react` in `@storybook/addon-storyshots`, so this "fix" is a filthy hack.

shilman committed:
Remove `@storybook/react` dependencies from storyshots

This is a hack. Currently the build is broken because `test-cra`
requires `@storybook/addon-storyshots` AND `@storybook/react`, but
`@storybook/addon-storyshots` requires `@storybook/react`. Before this
change, two versions of `@storybook/react` are getting loaded, and this
leads to a test error “Your test suite must contain at least one test.”

After this “fix” the error goes away. However,
`@storybook/addon-storyshots` should have a dev- dependency on
`@storybook/react`, because it uses it for its examples. But it only
has a peer-dep for now.

shilman committed:
Merge branch 'master' into 1266-fix-components-deps

shilman commented:
## Update again

I have fixed the build with an unfortunate hack. `@storybook/addon-storyshots` now only takes `@storybook/react` as a peer-dep, rather than as a dev dep.

This sucks, and is not technically correct, but it unbreaks the build. I am thinking about alternative approaches to this whole mess.

shilman committed:
Add placeholder tests to react-kitchen-sink

shilman committed:
Add package-lock.json

Per discussion with @tmeasday :wink:

tmeasday committed:
Use a pack -> install technique to recreate local packages

As suggested by https://github.com/npm/npm/pull/15900. This is a workaround until we get a better resolution of https://github.com/storybooks/storybook/issues/1286

tmeasday committed:
Added explanation of packs dir

tmeasday committed:
Revert removing `@storybook/react` from storyshots

shilman committed:
Merge pull request #1332 from storybooks/1331-test-cra-on-tarballs

Use a pack -> install technique to recreate local packages

shilman committed:
Create `@storybook/react/demo` exports

shilman committed:
Merge branch 'master' into 1266-fix-components-deps

shilman commented:
@alexandrebodin i missed your suggestion above about creating a `demo.js` file. i have incorporated that into this PR

shilman commented:
## Triumph!

@tmeasday 's PR uses tarballs for `test-cra`. This is a better solution than my hack above, so we backed that out. Merging.
